### PR TITLE
[luci-interpreter] Add Mean kernel

### PR DESCRIPTION
**Pull request by karthik-pen**
Add implementation of Mean kernel.

ONE-DCO-1.0-Signed-off-by: Karthik Penukonda <karthik.p@samsung.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Mean.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+void Mean::evalFloat() const
+{
+  const Shape input_shape = _input->shape();
+  int input_num_dims = input_shape.num_dims();
+  const auto *axis = getTensorData<int32_t>(_reduction_indices);
+  int num_axis = _reduction_indices->shape().num_elements();
+
+  // TODO
+  // Declare these temporary tensors somewhere else?
+  Tensor *temp_index = new Tensor(DataType::S32, Shape(input_num_dims), {}, "");
```
Comment (karthik-pen) [inline compiler/luci-interpreter/src/kernels/Mean.cpp:155]: I'm not sure where to declare and initialize these temporary tensors. Help needed.

**Comment by s-barannikov** [comment]
@karthik-pen Please do not create branches in upstream repo, create them in your clone.

**Comment by karthik-pen** [comment]
> 
> 
> @karthik-pen Please do not create branches in upstream repo, create them in your clone.

Ah, I didn't realize. My bad. I will be closing this PR and deleting this branch from upstream. I will create a new PR soon.

**Comment by karthik-pen** [comment]
Created new PR #1669 for this.

**Pull request closed without merging**